src/main/java/com/denizenscript/denizencore/utilities/CoreUtilities.java

```diff
@@ -709,6 +709,28 @@ public static String getClosestOption(List<String> strs, String opt) {
         return closest;
     }
 
+    public static int parseIndex(List list, String str) {
+        if(str.equalsIgnoreCase("last")) {
+            return list.size() - 1;
+        }
+        if(str.equalsIgnoreCase("first")) {
+            return 0;
+        }
+        int index;
+        try {
+            index = Integer.parseInt(str);
+        } catch (Exception ex) {
+            return -1;
+        }
+        if(index == 0) {
+            return -1;
+        }
+        if (index < 0) {
+            return list.size() + Math.max(1 - list.size(), index) - 1;
+        }
+        return Math.min(list.size(), index) - 1;

```

Comment:
Follow codebase formatting - space after `if`, if/else and try/catch blocks should have a newline separating them, e.g.
```java
if (cond) {

}
else {

}
```

Reply:
Ahh, I'll update those spaces, am I missing a spot where I didn't have the if/else separated by new line or is that just the catch you're referring to?

Reply:
Just the catch, I provided if/else as it's the most common example.